plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/TestOpenLineageListenerConfig.java

```diff
@@ -60,14 +62,39 @@ void testExplicitPropertyMappings()
                 .put("openlineage-event-listener.trino.include-query-types", "SELECT,DELETE")
                 .put("openlineage-event-listener.disabled-facets", "trino_metadata,trino_query_statistics")
                 .put("openlineage-event-listener.namespace", "testnamespace")
+                .put("openlineage-event-listener.job.name-format", "$QUERY_ID-$USER-$SOURCE-$CLIENT_IP-abc123")
                 .buildOrThrow();
 
         OpenLineageListenerConfig expected = new OpenLineageListenerConfig()
                 .setTrinoURI(new URI("http://testtrino"))
                 .setIncludeQueryTypes(ImmutableSet.of(SELECT, DELETE))
                 .setDisabledFacets(ImmutableSet.of(TRINO_METADATA, TRINO_QUERY_STATISTICS))
-                .setNamespace("testnamespace");
+                .setNamespace("testnamespace")
+                .setJobNameFormat("$QUERY_ID-$USER-$SOURCE-$CLIENT_IP-abc123");
 
         assertFullMapping(properties, expected);
     }
+
+    @Test
+    void testIsJobNameFormatValid()
+    {
+        assertThat(configWithFormat("abc123").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$QUERY_ID").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$USER").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$SOURCE").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$CLIENT_IP").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$QUERY_ID-$USER-$SOURCE-$CLIENT_IP-abc123").isJobNameFormatValid()).isTrue();
+        assertThat(configWithFormat("$QUERY_ID $USER $SOURCE $CLIENT_IP abc123").isJobNameFormatValid()).isTrue();
```

Comment:
Can we use `assertValidates` to verify them out

Reply:
Done